plugins/preparer/server_test.go

```diff
@@ -0,0 +1,94 @@
+package preparer
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/pkg/errors"
+)
+
+type testPreparer struct {
+	err               error
+	calledContextPath string
+	calledPluginData  map[string]string
```

Comment:
Do you want to add `calledCleanup bool` so you set that to `true` inside the `Cleanup` func and assert that it was called inside `Test_preparerServer_Cleanup`?